### PR TITLE
Prioritize explicit log lines over action logging

### PR DESCRIPTION
**Pull request by buoyad**
This changes the logger init to try to capture more explicit log lines. The specific logger decisions are somewhat arbitrary; the general idea is for actions to go in a smaller ring and explicit log lines to get logged natively / go in a larger ring. r? @keybase/react-hackers

### CODE REVIEW

File: shared/logger/index.js [hunk 1]
```diff
@@ -121,14 +120,18 @@ const devLoggers = () => ({
...
+    : new DumpPeriodicallyLogger(new RingLogger(200), 10 * 60e3, writeLogLinesToFile, 'Action'),
+  debug: new RingLogger(500),
   error: isMobile
     ? new NativeLogger()
     : new DumpPeriodicallyLogger(new RingLogger(100), 1 * 60e3, writeLogLinesToFile, 'Error'),
-  info: new RingLogger(100),
-  warn: new RingLogger(100),
+  info: isMobile
+    ? new NativeLogger()
+    : new DumpPeriodicallyLogger(new RingLogger(500), 1 * 60e3, writeLogLinesToFile, 'Info'),
```
Comment (akalin-keybase) [inline shared/logger/index.js]: error only has a ring size of 100, but info and warn has ring sizes of 500. Shouldn't error be at least as important as info/warn, and have a ring size of 500 or larger?

**Comment by zanderz** [comment]
Maybe in a future PR we could consider a category of logging that always goes in and isn't subject to being discarded if a ring fills up? Like we used to have? Also, the fact that the rings are separate in the uploaded log is problematic I think - in other words, they are ordered by ring first, actual time second, so you can't tell which order things happened in.

File: shared/logger/index.js (hunk 1 above)
Comment (buoyad) [reply shared/logger/index.js]: yep! my reasoning was that in practice we use warn a lot more than error, but that sounds fine to me

**Comment by buoyad** [comment]
@zanderz seems like we should have a layer (maybe when aggregating for a dump?) that inserts the log lines in the correct order. What do you think, @akalin-keybase?

**Comment by akalin-keybase** [comment]
@zanderz yeah I think for desktop we don't need any ring loggers at all, since we already do log rotation. That could be in this PR or a separate one.

**Comment by akalin-keybase** [comment]
@bouyad sgtm! one weirdness to watch out for is we currently debounce the log functions on native to avoid going over the bridge too much, but we take the timestamp after the debounce, so we'd likely want to capture the timestamp before the debounce, and then order by that.

**Comment by buoyad** [comment]
Ok, I'll ticket both of those

**Pull request merged**